src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -301,9 +438,181 @@ public Integer logData(
+            throw new ResponseStatusException( HttpStatus.BAD_REQUEST, "participant not found" );
+        }
+
+        // create association data
+        Map<UUID, Map<UUID, Set<Object>>> associationData = new HashMap<>();
+
+        associationDetails
+                .forEach( ( entityKeyId, entity ) -> {
+                    associationData.put( entityKeyId, new HashMap<>() );
+                    entity.forEach( ( propertyTypeFQN, data ) -> {
+                        UUID propertyTypeId = edmApi.getPropertyTypeId( propertyTypeFQN.getNamespace(),
+                                propertyTypeFQN.getName() );
+                        associationData.get( entityKeyId ).put( propertyTypeId, data );
+                    } );
+                } );
+
+        // update association entities
+        try {
+            dataApi.updateEntitiesInEntitySet( usedByEntitySetId, associationData, UpdateType.Replace );
+        } catch ( Exception exception ) {
+            logger.error( "error updating chronicle_used_by associations" );
+            throw new ResponseStatusException( HttpStatus.INTERNAL_SERVER_ERROR,
+                    "error updating chronicle_used_by associations" );
         }
 
-        dataApi.createEntityAndAssociationData( new DataGraph( entities, associations ) );
+        logger.info( "updated {} apps usage associations", associationDetails.size() );
+        return associationDetails.size();
+    }
+
+    // return a list of all the apps used by a participant filtered by the current date
+    @Override
+    public List<ChronicleAppsUsageDetails> getParticipantAppsUsageData(
+            UUID studyId,
+            String participantId ) {
+
+        logger.info( "Retrieving user apps: participantId = {}, studyId = {}", participantId, studyId );
+
+        SearchApi searchApi;
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            searchApi = apiClient.getSearchApi();
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to load apis" );
+            throw new ResponseStatusException( HttpStatus.INTERNAL_SERVER_ERROR );
+        }
+
+        UUID participantEntityKeyId = getParticipantEntityKeyId( participantId, studyId );
+        if ( participantEntityKeyId == null ) {
+            logger.error( "getUserApps: error retrieving participant. participant = {}, studyId = {}",
+                    participantId,
+                    studyId );
+            throw new ResponseStatusException( HttpStatus.BAD_REQUEST, "invalid participantId" );
+        }
+
+        UUID participantEntitySetId = getParticipantEntitySetId( studyId );
+        if ( participantEntitySetId == null ) {
+            logger.error( "getUserApps: error getting participant entity set id: participant = {}, studyId = {}",
+                    participantId,
+                    studyId );
+            throw new ResponseStatusException( HttpStatus.BAD_REQUEST, " invalid participantEntitySet" );
```

Comment:
Same comment. I feel like `BAD_REQUEST` and `"invalid participantEntitySet"` are not quite right. I feel like `INTERNAL_SERVER_ERROR` makes more sense with an error message similar to "unable to get the participant entity set id for the given study id".